### DISCUSSION TITLE
useMeta() is automatically available. How do your editor and ESLint know about it?

### DISCUSSION THREAD

**Discussion by elHornair** [nuxt/nuxt#18099] [Questions] [answered]
Hi friends!👋

I've noticed, that the `useMeta` composable is automatically available in the `setup` function (this is also mentioned in the [docs](https://v3.nuxtjs.org/docs/usage/meta-tags). The same seems to be true for Vue's reactivity helpers ('ref', 'reactive', 'watch' etc.).

This is cool, because it saves me work😅 My editor (Jetbrains/WebStorm) however, is not happy about that. It complains, that `useMeta` is not imported. Same for ESLint. How can I let them know, what dependencies are automatically available?

**Comment by elHornair** [comment]
Update: Just found [this](https://github.com/nuxt/framework/discussions/751#discussioncomment-1704901) fantastic demo by @Aperrix that solves the ESLint part. However the question about the Editor remains.

**Reply by Aperrix** [reply]
You can use a component as indicated in the documentation https://v3.nuxtjs.org/docs/usage/meta-tags#meta-components

**Reply by elHornair** [reply]
Thanks @Asperrix
This of course avoids the problem, but there are cases where you want to use `useMeta` instead of the meta-components. And the same problem exists with `ref`, `reactive`, `watch` etc.
How do you make your editor not complain about this?

**Reply by Aperrix** [reply]
Using volar with VScode, it's the best option as indicated in the documentation https://v3.nuxtjs.org/getting-started/introduction#prerequisites

**Reply by danielroe** [reply]
You can follow this issue: https://github.com/nuxt/framework/issues/2320

**Reply by elHornair** [reply]
Thanks @danielroe that clarified things 💪

**Comment by elHornair** [comment, marked as answer]
> You can follow this issue: #2320

This answers it
